src/all/mangapark/src/eu/kanade/tachiyomi/extension/all/mangapark/MangaParkDto.kt

```diff
@@ -38,33 +38,63 @@ class MangaParkComic(
     private val originalStatus: String? = null,
     private val uploadStatus: String? = null,
     private val summary: String? = null,
+    private val extraInfo: String? = null,
     @SerialName("urlCoverOri") private val cover: String? = null,
     private val urlPath: String,
+    @SerialName("max_chapterNode") private val latestChapter: Data<ImageFiles>? = null,
+    @SerialName("first_chapterNode") private val firstChapter: Data<ImageFiles>? = null,
 ) {
-    fun toSManga() = SManga.create().apply {
+    fun toSManga(shortenTitle: Boolean, pageAsCover: String) = SManga.create().apply {
         url = "$urlPath#$id"
-        title = name
-        thumbnail_url = cover
-        author = authors?.joinToString()
-        artist = artists?.joinToString()
-        description = buildString {
-            val desc = summary?.let { Jsoup.parse(it).text() }
-            val names = altNames?.takeUnless { it.isEmpty() }
-                ?.joinToString("\n") { "• ${it.trim()}" }
+        title = if (shortenTitle) {
+            name.replace(shortenTitleRegex, "").trim()
+        } else {
+            name
+        }
+        thumbnail_url = run {
+            val coverUrl = cover?.let {
+                when {
+                    it.startsWith("http") -> it
+                    it.startsWith("/") -> "https://$THUMBNAIL_LOOPBACK_HOST$it"
+                    else -> null
+                }
+            }
 
-            if (desc.isNullOrEmpty()) {
-                if (!names.isNullOrEmpty()) {
-                    append("Alternative Names:\n", names)
+            if (pageAsCover != "off" && useLatestPageAsCover(genres)) {
+                if (pageAsCover == "first") {
+                    firstChapter?.data?.imageFile?.urlList?.firstOrNull() ?: coverUrl
+                } else {
+                    latestChapter?.data?.imageFile?.urlList?.firstOrNull() ?: coverUrl
                 }
             } else {
-                append(desc)
-                if (!names.isNullOrEmpty()) {
-                    append("\n\nAlternative Names:\n", names)
-                }
+                coverUrl
             }
         }
+        author = authors?.joinToString()
+        artist = artists?.joinToString()
+        description = buildString {
+            if (shortenTitle) {
+                append(name)
+                append("\n\n")
+            }
+            summary?.also {
+                append(Jsoup.parse(it).wholeText().trim())
+                append("\n\n")
+            }
+            extraInfo?.takeUnless(String::isBlank)?.also {
+                append("Extra Info:\n")
+                append(Jsoup.parse(it).wholeText().trim())
+                append("\n\n")
+            }
+            altNames?.takeUnless { it.isEmpty() }
```

Comment:
Why mix syntax?
```kt
.takeUnless(String::isBlank)
.takeUnless { it.isEmpty() }
```
I prefer the first one, looks less busy. But should at least stay consistent with whatever approach you pick.